api/compute/dataset_persist.go

```diff
@@ -264,31 +264,84 @@ func splitTrainTest(sourceFile string, trainFile string, testFile string, hasHea
 		rowData = append(rowData, line)
 	}
 
+	if stratify {
+		// For statification we use a proportionate allocation strategy, dividing the dataset up into
+		// subsets by category, and then sampling the subsets using the supplied train/test ratio.
+
+		// first pass - create subsets by category
+		categoryRowData := map[string][][]string{}
+		for _, row := range rowData {
+			key := row[targetCol]
+			if _, ok := categoryRowData[key]; !ok {
+				categoryRowData[key] = [][]string{}
+			}
+			categoryRowData[key] = append(categoryRowData[key], row)
+		}
+
+		// second pass - randomly sample each category to generate train/test split
+		for _, data := range categoryRowData {
+			maxCategoryRows := len(rowData) / len(data) * maxTrainingCount
```

Comment:
I think this division is reversed as it is currently going to set max cat rows to a multiple of max training count

Reply:
Fixed.